contribs/gmf/src/directives/search.js

```diff
@@ -0,0 +1,277 @@
+/**
+ * @fileoverview this file provides the "search" directive and controller for
+ * a GeoMapFish application.
+ *
+ * Example:
+ *
+ * <gmf-search gmf-search-map="ctrl.map"></gmf-search>
+ */
+goog.provide('gmf.searchDirective');
+
+
+goog.require('gmf');
+goog.require('gmf.mapDirective');
+goog.require('ngeo.CreateGeoJSONBloodhound');
+goog.require('ngeo.searchDirective');
+goog.require('ol.Map');
+goog.require('ol.View');
+goog.require('ol.layer.Tile');
+goog.require('ol.layer.Vector');
+goog.require('ol.proj');
+goog.require('ol.source.OSM');
+goog.require('ol.source.Vector');
+
+
+/**
+ * @return {angular.Directive} The Directive Definition Object.
+ * @ngInject
+ */
+gmf.searchDirective = function() {
+  return {
+    restrict: 'E',
+    scope: {
+      'getMapFn': '&gmfSearchMap',
+      'getDatasourcesFn': '&gmfSearchDatasources'
+    },
+    controller: 'GmfSearchController',
+    controllerAs: 'ctrl',
+    template:
+        '<input type="text" placeholder="search…" ' +
+        'ngeo-search="ctrl.options" ' +
+        'ngeo-search-datasets="ctrl.datasets" ' +
+        'ngeo-search-listeners="ctrl.listeners">',
+    link:
+        /**
+         * @param {angular.Scope} scope Scope.
+         * @param {angular.JQLite} element Element.
+         * @param {angular.Attributes} attrs Atttributes.
+         */
+        function(scope, element, attrs) {
+          // Empty the search field on focus and blur.
+          element.find('input').on('focus blur', function() {
+            $(this).val('');
+          });
+        }
+  };
+};
+
+gmfModule.directive('gmfSearch', gmf.searchDirective);
+
+
+
+/**
+ * @constructor
+ * @param {angular.Scope} $scope The directive's scope.
+ * @param {angular.$compile} $compile Angular compile service.
+ * @param {ngeo.CreateGeoJSONBloodhound} ngeoCreateGeoJSONBloodhound The ngeo
+ * create GeoJSON Bloodhound service.
+ * @ngInject
```

Comment:
You need an `@export` annotation here. See https://github.com/camptocamp/ngeo/blob/master/docs/guidelines.md#use-export-for-controllers if you want to understand why.